File: ethal/ethal/report/qc/qc.py
# Copyright (c) 2013, Atrina Technologies Pvt. Ltd. and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data = [], []
	columns = ["Item Name:Link/Item:250"]+["Rejection Ash:Float:100"]+["Rejection Blister:Float:100"]+["Rejection Rolling:Float:100"]+["Rejection Others:Float:100"]+["OK Circle Received:Float:100"]+["OK Circle Received(Pieces):Int:100"]+["Total Rejection:Float:100"]+["Melting Rejection:Float:100"]+["Total Circle Received:Float:100"]
	data = get_data(filters)
	return columns, data

def get_data(filters):
	return frappe.db.sql("""
		select item_name, rejection_ash, rejection_blister, rejection_rolling, rejection_others, ok_circle_received,
			ok_circle_receivedpieces, total_rejection, melting_rejection, total_circle_received
		from `tabQC Items` 
		where parent = '{0}'
		order by idx	
	""".format(filters.qc))

File: ethal/ethal/report/qc/qc.js
// Copyright (c) 2016, Atrina Technologies Pvt. Ltd. and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["QC"] = {
	"filters": [
		{
			"fieldname":"qc",
			"label": __('QC'),
			"fieldtype": "Link",
			"options": "QC",
			// "on_change": function(query_report) {
			// 	var name = query_report.get_values().qc
			// 	console.log(name)
			// 	frappe.db.get_list('QC', {
			// 		fields: ['date', 'day', 'month'],
			// 		filters: {
			// 			name: name
			// 		}
			// 	}).then(records => {
			// 		console.log(records[0]);
			// 		frappe.query_report.set_filter_value('date', records[0].date)
			// 		frappe.query_report.set_filter_value('day', records[0].day)
			// 		frappe.query_report.set_filter_value('month', records[0].month)
			// 	})
			// }		
		},
		{
			"fieldname":"from_date",
			"label": __('From Date'),
			"fieldtype": "Date"
		},
		{
			"fieldname":"to_date",
			"label": __('To Date'),
			"fieldtype": "Date"
		},
		// {
		// 	"fieldname":"day",
		// 	"label": __('Day'),
		// 	"fieldtype": "Data",
		// 	"read_only": 1
		// },
		// {
		// 	"fieldname":"month",
		// 	"label": __('Month'),
		// 	"fieldtype": "Data",
		// 	"read_only": 1
		// }
	]
};
